src/detect-transform-dropbytes.c

```diff
@@ -0,0 +1,181 @@
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_TRANSFORM_DROPBYTES].RegisterTests = DetectTransformDropbytesRegisterTests;
+#endif
+    sigmatch_table[DETECT_TRANSFORM_DROPBYTES].flags |= SIGMATCH_QUOTES_MANDATORY;
+}
+
+static void DetectTransformDropbytesFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    if (ptr != NULL) {
+        DetectTransformDropbytesData *pdd = (DetectTransformDropbytesData *)ptr;
+        SCFree(pdd->bytes_to_drop);
+        SCFree(pdd);
+    }
+}
+
+/**
+ *  \internal
+ *  \brief Apply the dropbytes keyword to the last pattern match
+ *  \param det_ctx detection engine ctx
+ *  \param s signature
+ *  \param optstr options string
+ *  \retval 0 ok
+ *  \retval -1 failure
+ */
+static int DetectTransformDropbytesSetup(DetectEngineCtx *de_ctx, Signature *s, const char *optstr)
+{
+    SCEnter();
+
+    // Create pdd from optstr
+    DetectTransformDropbytesData *pdd = SCCalloc(1, sizeof(*pdd));
+    if (pdd == NULL) {
+        SCLogError("memory allocation failed");
+        SCReturnInt(-1);
+    }
+
+    char *comma;
+    int ret = 0;
+    uint8_t *content = NULL;
+    uint16_t len = 0;
+
+    pdd->clear = false;
+    pdd->negate = false;
+
+    comma = strchr(optstr, ',');
+
+    if (comma == NULL) {
+        ret = DetectContentDataParse("dropbytes", optstr, &content, &len);
+    } else {
+        for (uint32_t i = 0; i < (comma - optstr); i++) {
+            if (optstr[i] == 'c') {
+                pdd->clear = true;
+            } else if (optstr[i] == '!') {
+                pdd->negate = true;
+            } else {
+                SCLogError("DROPBYTES unknown flag: %c", optstr[i]);
+                DetectTransformDropbytesFree(de_ctx, pdd);
+                SCReturnInt(-1);
+            }
+        }
+        ret = DetectContentDataParse("dropbytes", comma + 1, &content, &len);
+    }
+
+    if (ret == -1 || content == NULL) {
+        SCLogError("DROPBYTES option error");
+        DetectTransformDropbytesFree(de_ctx, pdd);
+        SCReturnInt(-1);
+    }
+
+    pdd->bytes_to_drop = content;
+    pdd->cnt_bytes_to_drop = len;
+    int r = DetectSignatureAddTransform(s, DETECT_TRANSFORM_DROPBYTES, pdd);
+    if (r != 0) {
+        DetectTransformDropbytesFree(de_ctx, pdd);
+    }
+
+    SCReturnInt(r);
+}
+
+static void DetectTransformDropbytes(InspectionBuffer *buffer, void *options)
+{
+    const uint8_t *input = buffer->inspect;
+    const uint32_t input_len = buffer->inspect_len;
+    uint32_t output_len = 0;
+    DetectTransformDropbytesData *pdd = options;
+    if (input_len == 0) {
+        return;
+    }
+    uint8_t output[input_len];
+
+#ifdef DEBUG
+    uint8_t dump[10000];
+    uint32_t offset = 0;
+    PrintRawDataToBuffer(dump, &offset, 10000, input, input_len);
+    SCLogDebug("Input:\n%s\n", (char *)dump);
+#endif
+
+    for (uint32_t i = 0; i < input_len; i++) {
+        bool keep = !pdd->negate;
+        for (uint32_t k = 0; k < pdd->cnt_bytes_to_drop; k++) {
```

Comment:
Instead of having this loop, could `DetectTransformDropbytesData` have a `bool[256]` to know if needs to keep a character : more ram consumption, but faster CPU execution ?